src/generator.ts

```diff
@@ -158,9 +166,21 @@ function generateType(ast: AST, options: Options): string {
     case 'OBJECT': return 'object'
     case 'REFERENCE': return ast.params
     case 'STRING': return 'string'
-    case 'TUPLE': return '['
-      + ast.params.map(_ => generateType(_, options)).join(', ')
+    case 'TUPLE': return (() => {
+      const params = ast.params.map(_ => generateType(_, options))
+      if (ast.spreadParam) {
+        let spread = generateType(ast.spreadParam, options)
+        if (spread.endsWith('"')) {
+          spread = '(' + spread + ')'
```

Comment:
Nit: Consider always adding the `()` for simplicity, and letting the prettifier strip them out for you if they're not necessary.

Reply:
Interestingly, prettier doesn't actually remove them if they're completely unnecessary.

[prettier repl](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACAS1jmwDMBDMOPAQW23IE8aATFgmA6cgGwEl4AWwDOeYAB0oeaTKjlBcFgFEAriID8SMZJm685Nhy5z+Q4TSiNNeANoA5eYqVQ1AJgA0eAHQ+AFOUsAShsAXRCAbh09aQN2Tm5TOA0te0dlF0FXAEZPHy9fBwV0t1dgsMipPQBfCuiVKDkipUE1YWsJSujYowSBJPNLaxtfLLwAHzxXcbwAZkDcvwDGMoiorsN4kz7k219xEAB5KDh96f2AFQB3CFOJi4ALbDgTkHnvPwAWaYBWaYA2Fa1apA3SNRQAZTA9yS5Haaz03U2vG2AysKXBMGwRAA5pDoYJyDk8HY1AAjEh4mFEvL+IKhVadBEbYzIsxDDFYqC4qEwjzEskUnkEvk0gBCEAgPDgAUpBKygPh0hqirwwiFsK0HWiMWZvTMFjRthVui12t0pAl1mEmJxILNtgA1lprZzsSEtEs7dEqu5jTJTfa8KTyNhrBlydgvdqbE7VTaue79JYo7ofX7pDSlmU-QyzYiWYkdjZ09pGWaLRArfHsSm9DHndXE56S2my3oA-bg6GtOGSLXdPW466m8mW7627oaR2zcGAF7WUkSqUBfsyQcunEjxirqrZid4XPAlUQUjnRgYOBw-eIvX9A1DXvYTwbrkLfJZ+m1m9bNkpR-P6s318RdJWlKAFTbZU2yeUg2k1P181vER7xSJQ0HkDApVlcggI-coEN1H9+iGNCMKw9VqT8UjBEwuBsNKT8VSg6IWDgWCrzzQjWTvQYUgAMQIbBrQAETYogeigXC6Xw68uMLODbAAVSgJ4eHIeAWHODgpSA857kElhRNIcTNgg707SqeF0CwXBCGIMhKGoDkcWw0ZpzwCsq1dKN10bD1R0ZSzGWsnB8CIeAHKoflBAjVzS2iLswwFSMVV84d-O3eEgt0ELbPCkgKCi5yuXo+K9E8htvNS2MXzdDKLKszBQrsiLCuoEkYsFfFyCmdzEp7ZKfJqvyk0ywLGpssL7LavBxVAmUKLK3Q5wXJcwKGyrN3qrKJua-LIuoajaOwpaZGMoSYEKOBNq5KM1OtK6bprFUAHoACo3uNN7aGxagiDwRhpSEr6XpVchfqSzqUrbaECGxe4YEhiMowoAA3HAODgPiJRYBSllCO6CAdOBhGEiBsQUkDlygDahy20aGuCpq8umxy8CO8jurclVzoexwnru8g+YUAXXo+r6fr+qRAZDYQQbBiGBqhqNYfhxGleRnnyHRrF4GxiBcesfHD10HgiZJsmKdW+aaeqp6t0ZnLmam1q2Y5uj1V6lUCGEABhaBSDNsAYC0Kn1p2xl3s+stvq0mApTwAkYB5URWOMqAJLwYnGA8nAs4IMAHTlmPQaZyaWoKtmBIuoyTOMU7pDBLzbQj3Qo8VWPtL+0R6lU9TFDwGAIDwNO6+gLO4BzgIWCDOB7m1rgVGwQhRBgaFVUcDvS6d8v9pm5S+40uOE-cpvRfGyPxZjvBa4zzY8Hn0QoGHzh4+oae8AwbADZUKgn+HlgKhMIF37ngPgrMqBb12izV2UU9IGVvpnU+jhm63VbjIaB+BIBQGtHgBAahiRpGcG4BueAjjUAALx4ByCqK4w8qEeFoY8Z4eAqEzHQdIXKWDoC4PwYIQhTQMjZFIeQ1h1DxzRDoWIxhbY9JPEobMDheDnZ4GwbwjIAinBCK9m2bGS8xEfAkXoASqMFHfCMbocEBA0BiL+FlEA7gQAQAwJsYQyBQAhm-pcAACiGBAbiUDawgAQFgDiQCkgYIXOAMBwQYEoDiZAmIVBwEcQjQQPAADq+l4DCDiVQcE-ijCow4IwZA4BhBuMcUQYQJAYDeIYNiAkyAKA8BqY4gAVsINAopInExiY4AAMkQOAzTeBtJAJ0tAViuRSgAIoqAgPAUZrSUkgDiUJEgZSYDnhJmALELiwlf3ChkkJa9kAAA4AAMjiv4QBqRkhgGAylfxJiQUxYSACOCz4D1OcQEkAQsAC0xxFCKDCU8L5gk4D1PBk0pALTxk1MEAQRJ2BkmOOEDiOZ3yRnwrGasmA5BSQnJYGcpAHgQCYnIAQM2XJ-aCDhSAZ+xwwkqBqecIlASEVwCqFUIAA)